pkg/xds/envoy/listeners/v3/rbac_configurer_test.go

```diff
@@ -0,0 +1,244 @@
+package v3_test
+
+import (
+	listenerv3 "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	core_xds "github.com/kumahq/kuma/pkg/core/xds"
+	"github.com/kumahq/kuma/pkg/plugins/policies/meshtrafficpermission/api/v1alpha1"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+	"github.com/kumahq/kuma/pkg/xds/envoy"
+	"github.com/kumahq/kuma/pkg/xds/envoy/listeners"
+	v3 "github.com/kumahq/kuma/pkg/xds/envoy/listeners/v3"
+)
+
+var _ = Describe("RBACConfigurer", func() {
+
+	type testCase struct {
+		rules    core_xds.Rules
+		mesh     string
+		stats    string
+		expected string
+	}
+
+	DescribeTable("should generate proper envoy config",
+		func(given testCase) {
+			// given
+			configurer := &v3.RBACConfigurer{
+				Rules:     given.rules,
+				Mesh:      given.mesh,
+				StatsName: given.stats,
+			}
+			res, err := listeners.NewFilterChainBuilder(envoy.APIV3).Build()
+			Expect(err).ToNot(HaveOccurred())
+
+			// when
+			err = configurer.Configure(res.(*listenerv3.FilterChain))
+			Expect(err).ToNot(HaveOccurred())
+
+			// then
+			actual, err := util_proto.ToYAML(res)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(actual).To(MatchYAML(given.expected))
+		},
+		Entry("allow all", testCase{
+			stats: "allow_all_prefix",
+			mesh:  "allow_all_mesh",
+			rules: []*core_xds.Rule{
+				{
+					Subset: []core_xds.Tag{},
+					Conf: &v1alpha1.MeshTrafficPermission_Conf{
+						Action: "ALLOW",
+					},
+				},
+			},
+			expected: `
+filters:
+  - name: envoy.filters.network.rbac
+    typedConfig:
+      '@type': type.googleapis.com/envoy.extensions.filters.network.rbac.v3.RBAC
+      rules:
+          policies:
+              MeshTrafficPermission:
+                  permissions:
+                      - any: true
+                  principals:
+                      - any: true
+      statPrefix: allow_all_prefix.`,
+		}),
+		Entry("deny all", testCase{
+			stats: "deny_all_prefix",
+			mesh:  "deny_all_mesh",
+			rules: []*core_xds.Rule{
+				{
+					Subset: []core_xds.Tag{},
+					Conf: &v1alpha1.MeshTrafficPermission_Conf{
+						Action: "DENY",
+					},
+				},
+			},
```

Comment:
Is there no way where there's no Rule at all?

Reply:
Sorry, I'm not sure I'm following the question

Reply:
Should there be a testCase:

```go
testCase{
  stats: "no",
  mesh: "nothing",
  rules: []*core_xds.Rule{Subset: []core_xds.Tag{}},
```

Or something like this?

Reply:
oh I get it, added a testCase `no rules`